crates/starknet-devnet-core/src/starknet/mod.rs

```diff
@@ -490,20 +490,21 @@ impl Starknet {
             },
         };
 
-        BlockContext::new_unchecked(&block_info, &chain_info, &get_versioned_constants())
+        BlockContext::new(block_info, chain_info, get_versioned_constants(), BouncerConfig::max())
     }
 
     /// Update block context block_number with the next one
     /// # Arguments
     /// * `block_context` - BlockContext to be updated
     fn advance_block_context_block_number(block_context: &mut BlockContext) {
         let mut block_info = block_context.block_info().clone();
-        block_info.block_number = block_info.block_number.next();
+        block_info.block_number = block_info.block_number.next().unwrap_or_default();
         // TODO: update block_context via preferred method in the documentation
-        *block_context = BlockContext::new_unchecked(
-            &block_info,
-            block_context.chain_info(),
-            &get_versioned_constants(),
+        *block_context = BlockContext::new(
+            block_info,
+            block_context.chain_info().clone(),
+            get_versioned_constants(),
+            BouncerConfig::max(),
```

Comment:
I saw it as the safest solution for now. The best would be to allow users to define their own config, but it would introduce around 15 new CLI options (or perhaps one if we are to introduce one big ugly parameter that will accept the whole config, or something that will accept a path to a separate config file). Alternatively we can just hardcode some values (though it might also make sense to read the from the origin in case of forking).

Reply:
I prefer to be defined in some way to reflect testnet and/or mainnet but for some of the fields I can't find any information. https://docs.starknet.io/tools/limits-and-triggers/

Reply:
I can propose to set all fields that you have information for and the others to be set to max value

Reply:
Set gas and n_steps to the values from the page, the rest to max: https://github.com/0xSpaceShard/starknet-devnet-rs/pull/557/commits/cb422b5262bb9150bba2c3d3b3694266d713080f